crates/core/app/tests/mock_consensus.rs

```diff
@@ -56,12 +63,30 @@ async fn mock_consensus_can_send_a_single_empty_block() -> anyhow::Result<()> {
             .await?
     };
 
-    engine
-        .block()
-        .with_data(vec![])
-        .with_evidence(List::new(Vec::new()))
-        .execute()
-        .await?;
+    // Check that we begin at height 0, before any blocks have been generated.
+    assert_eq!(
+        storage.latest_snapshot().get_block_height().await?,
+        0,
+        "height should begin at 0"
+    );
+
+    for expected in 1..=8 {
+        // Generate an empty block.
+        engine
+            .block()
+            .with_data(vec![])
+            .with_evidence(List::new(Vec::new()))
```

Comment:
These don't need to be set, right? Wondering if this is an artefact of the dev process, as a (potential) user it's confusing to see invocations of methods with default parameters.

Reply:
this is a great catch, thanks. #3848 makes the evidence optional, using an empty list if none is provided.